dvc/repo/__init__.py

```diff
@@ -370,9 +371,17 @@ def used_cache(
             to items containing the output's `dumpd` names and the output's
             children (if the given output is a directory).
         """
-        from dvc.objects.db import NamedCache
+        used_objs = set()
+        used_external = defaultdict(set)
 
-        cache = NamedCache()
+        def _add_suffix(objs, suffix):
+            from dvc.objects.tree import Tree
+
+            for obj in objs:
+                obj.name += suffix
+                if isinstance(obj, Tree):
+                    for _, entry_obj in obj:
+                        entry_obj.name += suffix
```

Comment:
This workaround is needed to preserve current `status -c` and `repo.status()` behavior.

Assuming that we want to preserve the existing `status -c` behavior going forward, it seems like what we will is for `repo.status()` to handle returning  the mapping of objects or object IDs to the set of revision+path pairs that point to each object, similar to what NamedCache did before.

But it should be handled at the repo level rather than passing string suffixes all the way down to the output `collect_used_cache` methods like we were doing before.